hardware_interface/test/fake_components/test_generic_system.cpp

```diff
@@ -323,8 +358,78 @@ class TestGenericSystem : public ::testing::Test
   std::string hardware_system_2dof_standard_interfaces_with_offset_;
   std::string hardware_system_2dof_standard_interfaces_with_custom_interface_for_offset_;
   std::string hardware_system_2dof_standard_interfaces_with_custom_interface_for_offset_missing_;
+  std::string valid_urdf_ros2_control_system_robot_with_gpio_;
 };
 
+TEST_F(TestGenericSystem, valid_urdf_ros2_control_system_robot_with_gpio_)
+{
+  auto urdf = ros2_control_test_assets::urdf_head + valid_urdf_ros2_control_system_robot_with_gpio_ +
+              ros2_control_test_assets::urdf_tail;
+  hardware_interface::ResourceManager rm(urdf);
+
+  // check is hardware is configured
+  std::unordered_map<std::string, rclcpp_lifecycle::State> states_map;
+  states_map = rm.get_components_states();
+  EXPECT_EQ(
+    states_map["GenericSystem2dof"].label(), hardware_interface::lifecycle_state_names::INACTIVE);
+
+  // Check initial values
+  hardware_interface::LoanedStateInterface gp1_s = rm.claim_state_interface("flange_analog_IOs/analog_output1");
+  hardware_interface::LoanedStateInterface gp2_s = rm.claim_state_interface("flange_vacuum/vacuum");
+  hardware_interface::LoanedCommandInterface gp1_c = rm.claim_command_interface("flange_analog_IOs/analog_output1");
+  hardware_interface::LoanedCommandInterface gp2_c = rm.claim_command_interface("flange_vacuum/vacuum");
```

Comment:
Please also add here claim of the other states `analog_input1` and `analog_input2`?

Then we would need some kind of "fake command interfaces" (see impl. for sensors) to control those from the "outside". You probably need to add a new flag for this.

Reply:
I need to check this once.